Fix usb_cdc.enable(console=False, data=True)

dhalbert opened the pull request:
`usb_cdc.enable(console=False, data=True)` was not working.

- Set `usb_cdc` defaults correctly
- Use correct TinyUSB `itf` values when only one CDC is used. Number them from zero. This was the main bug.
- Don't watch for ctrl-c if console is not enabled.
- add `board.LED` to Metro M4. (Discovered it was missing by accident when i was trying to blink the led for debugging purposes).'

Tested on Feather STM32F405 and on Metro M4.

Thanks to discord user osten (@pdecherf) who found the problem and tested the fix on an STM32F405.

jepler reviewed (approved):
Makes sense, but no testing performed.  Are there more places we should switch to `tud_cdc_n`, in the long term?

dhalbert commented:
> Makes sense, but no testing performed. Are there more places we should switch to `tud_cdc_n`, in the long term?

I looked in all the places to understand them, and added comments. Mainly, I fixed the broken simmel build.

jepler reviewed (approved):
thanks!  once again, no testing performed.
